src/Storages/MergeTree/MergeTask.cpp

```diff
@@ -122,6 +122,20 @@ static void addMissedColumnsToSerializationInfos(
     }
 }
 
+bool MergeTask::GlobalRuntimeContext::isCancelled() const
+{
+    return merges_blocker->isCancelledForPartition(future_part->part_info.partition_id)
```

Comment:
I assume here that the merge operation doesn't create parts in other partitions, so we might check only if the destination partition was canceled or not.